src/analysis/util.cc

```diff
@@ -25,131 +25,98 @@ const std::vector<Piece> kPieceSet{
-
-    hand.pop_back();
   }
   return waits;
 }
 
-const int kPiecesinahand = 14;
+std::vector<Piece> getWaits(const Hand& hand, const Piece& piece) {
+  Hand new_hand = hand;
+  new_hand.live.erase(
+      std::ranges::find(new_hand.live, piece));
+  return getWaits(new_hand);
+}
 
-std::vector<Piece> isInTenpai(std::vector<Piece> hand, bool allWaits) {
-  if (hand.empty()) {
-    return {};
-  }
-  const int min_singles = countSingles(hand);
-  // These numbers are the same as above except one more piece means 1 higher on the single count
-  if (min_singles > 6 || min_singles == 1 || min_singles == 4 ||
-      min_singles == 0) {
+std::map<Piece, std::vector<Piece>> getPossibleWaits(const Hand& hand) {
+  // 14, 11, 8, 5, 2 are the possible piece counts for non-melded tiles.
+  if (hand.live.empty() || hand.live.size() % 3 != 2) {
     return {};
   }
-  std::array<bool, Piece::kPiecesize> removedbefore = {};
-  std::vector<Piece> waits;
-  for (int i = 0; i < kPiecesinahand; i++) {
-    const Piece removed = hand.front();
-    hand.erase(hand.begin());
-    if (removedbefore.at(removed.toUint8_t())) {
-      hand.push_back(removed);
+  constexpr int kMaxSingles = 3;
+  std::map<Piece, std::vector<Piece>> waits;
+  for (const Branch& branch : AnalyzeHand(hand)) {
+    if (branch.singles.size() > kMaxSingles) {
       continue;
     }
-    removedbefore.at(removed.toUint8_t()) = true;
-    std::vector<Piece> tempwaits = isInTenpai13Pieces(hand, allWaits);
-    if (!tempwaits.empty()) {
-      if (!allWaits) {
-        return tempwaits;
-      }
-      waits.insert(waits.begin(), tempwaits.begin(), tempwaits.end());
+    if (branch.singles.size() == 2 &&
+        (branch.pairs.empty() || branch.pairs.size() == 6)) {
+      waits[branch.singles[0]].emplace_back(branch.singles[1]);
+      waits[branch.singles[1]].emplace_back(branch.singles[0]);
+    } else if (branch.singles.size() == 3) {
+      waits[branch.singles[0]].append_range(
+          completeSet(branch.singles[1], branch.singles[2]));
+      waits[branch.singles[1]].append_range(
+          completeSet(branch.singles[2], branch.singles[0]));
+      waits[branch.singles[2]].append_range(
+          completeSet(branch.singles[0], branch.singles[1]));
     }
-    hand.push_back(removed);
   }
   return waits;
 }
 
-int countSingles(const std::vector<Piece>& hand) {
-  auto root = breakdownHand(hand);
-  int min_singles = 15;
-  for (const auto& branch : Node::AsBranchVectors(root.get())) {
-    int singles = 0;
-    for (const auto* node : branch) {
-      if (node->type() == SetType::kSingle) {
-        singles++;
-      }
-    }
-    min_singles = std::min(singles, min_singles);
-  }
-  return min_singles;
-}
-
-int countPiece(const GameState& state, int player, Piece p) {
-  int count = 0;
-  for (const auto& piece : state.hands.at(player).live) {
-    if (piece == p) {
-      count++;
-    }
-  }
-  return count;
+bool CheckBranch(const Branch& branch, const SetCheckFunc& func) {
```

Comment:
Can you remind me the function naming convention for PascalCase vs camel case here?